traffic_ops/app/db/trafficvault/scripts/fill_expiration_and_provider.go

```diff
@@ -0,0 +1,217 @@
+
+	if *help {
+		flag.Usage()
+		os.Exit(0)
+	}
+
+	aesKey, err := readKey(*aesKeyLocation)
+	if err != nil {
+		die("reading previous-key: " + err.Error())
+	}
+
+	dbConfBytes, err := ioutil.ReadFile(*cfg)
+	if err != nil {
+		die("reading db conf '" + *cfg + "': " + err.Error())
+	}
+
+	pgCfg := Config{}
+	err = json.Unmarshal(dbConfBytes, &pgCfg)
+	if err != nil {
+		die("unmarshalling '" + *cfg + "': " + err.Error())
+	}
+
+	sslStr := "require"
+	if !pgCfg.SSL {
+		sslStr = "disable"
+	}
+
+	db, err := sqlx.Open("postgres", fmt.Sprintf("postgres://%s:%s@%s:%d/%s?sslmode=%s&fallback_application_name=trafficvault", pgCfg.User, pgCfg.Password, pgCfg.Hostname, pgCfg.Port, pgCfg.DBName, sslStr))
+	if err != nil {
+		die("opening database: " + err.Error())
+	}
+
+	tx, err := db.Begin()
+	if err != nil {
+		die(fmt.Sprintf("transaction begin failed %v %v ", err, tx))
+	}
+	defer tx.Commit()
+
+	rows, err := tx.Query("SELECT deliveryservice, cdn, version, data, provider, expiration FROM sslkey")
+	if err != nil {
+		die("querying: " + err.Error())
+	}
+	defer rows.Close()
+
+	type expiryAndProvider struct {
+		Provider   string
+		Expiration time.Time
+	}
+	sslKeyMap := map[string]expiryAndProvider{}
+
+	for rows.Next() {
+		var ds string
+		var cdn string
+		var version string
+		var encryptedSslKeys []byte
+		provider := sql.NullString{}
+		var expiration time.Time
+		if err = rows.Scan(&ds, &cdn, &version, &encryptedSslKeys, &provider, &expiration); err != nil {
+			die("getting SSL Keys: " + err.Error())
+		}
+		id := strings.Join([]string{ds, cdn, version}, ", ")
+		jsonKeys, err := util.AESDecrypt(encryptedSslKeys, aesKey)
+		if err != nil {
+			die("reading SSL Keys: " + err.Error())
+		}
+
+		if !bytes.HasPrefix(jsonKeys, []byte("{")) {
+			die("decrypted SSL Key did not have prefix '{' for id " + id)
+		}
+
+		sslKey := tc.DeliveryServiceSSLKeysV15{}
+		err = json.Unmarshal([]byte(jsonKeys), &sslKey)
+		if err != nil {
+			die("unmarshalling ssl keys: " + err.Error())
+		}
+
+		parsedCert := sslKey.Certificate
+		err = Base64DecodeCertificate(&parsedCert)
+		if err != nil {
+			die("getting SSL keys for ID '" + id + "': " + err.Error())
+		}
+
+		block, _ := pem.Decode([]byte(parsedCert.Crt))
+		if block == nil {
+			die("Error decoding cert to parse expiration")
+		}
+
+		x509cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			die("Error parsing cert to get expiration - " + err.Error())
+		}
+
+		sslKeyMap[id] = expiryAndProvider{
+			Provider:   sslKey.AuthType,
+			Expiration: x509cert.NotAfter,
+		}
+	}
+
+	for id, info := range sslKeyMap {
+		idParts := strings.Split(id, ", ")
+		ds := idParts[0]
+		cdn := idParts[1]
+		version := idParts[2]
+		res, err := tx.Exec(`UPDATE sslkey SET provider = $1, expiration = $2 WHERE deliveryservice = $3 AND cdn = $4 AND version = $5`, info.Provider, info.Expiration, ds, cdn, version)
+		if err != nil {
+			die(fmt.Sprintf("updating SSL Keys for %s, %s", id, err.Error()))
```

Comment:
nit but calling `.Error()` is already the default string coercion for errors, so if you know it isn't `nil` `%s` formatting suffices: https://play.golang.org/p/FtjDb-2ze0Y (if it might be `nil` you'd need to use `%v` instead to avoid segfaulting in that case - but still don't need to explicitly call `.Error()`).